### ISSUE TITLE
Create a threshold module

### ISSUE THREAD

**Issue by cpehle** [norse/norse#207]
Currently we handle threshold parametrisation by dispatching on a string.
This was originally done to support JIT specialisation. We should be able to accomplish the same by
using a torch.nn.Module which captures the necessary parameters of the chosen
threshold detection / pseudo-gradient computation.

**Comment by Jegp** [comment]
What's exactly the purpose of this? Performance? Useability?
